sdk/storage/azblob/pageblob/client.go

```diff
@@ -150,14 +150,18 @@ func (pb *Client) Create(ctx context.Context, size int64, o *CreateOptions) (Cre
 // This method panics if the stream is not at position 0.
 // Note that the http client closes the body stream after the request is sent to the service.
 // For more information, see https://docs.microsoft.com/rest/api/storageservices/put-page.
-func (pb *Client) UploadPages(ctx context.Context, body io.ReadSeekCloser, options *UploadPagesOptions) (UploadPagesResponse, error) {
+func (pb *Client) UploadPages(ctx context.Context, body io.ReadSeekCloser, contentRange blob.HTTPRange, options *UploadPagesOptions) (UploadPagesResponse, error) {
```

Comment:
From the internal discussion.

> We are taking ReadSeekCloser as input and calculating length of it in code. If no range is given then we are overwriting from 0 offset so shall be ok.
that's actually not correct I attempted to have 0 and the API fails - it requires a proper range, I have looked at python as well and range is expected to be provided explicitly - so the customer needs to have it handy

It's not clear to me if `HTTPRange` is really required or we have a bug in our math to calculate the range from the provided `ReadSeekCloser`.  Can somebody clarify?

CC @JeffreyRichter

Reply:
For UploadPages, a start offset MUST be specified by the caller. The stream has to be a multiple of 512 bytes and we could calculate the end by adding the stream length to the start offset.

Reply:
I see.  Are there cases where the caller might want to specify an end that's shorter than the `ReadSeekCloser`?  If so, then specifying an explicit `HTTPRange` makes sense.

Reply:
Let's just keep the `HTTPRange` considering this is the power-user scenario where complete control is best.

Reply:
Either customer specifies the end (and we don't read the whole stream) or the customer wraps the stream with a call to https://pkg.go.dev/io#LimitReader. I kind like the latter because the customer knows what to expect of the stream but LimitReader returns an io.Reader; not an io.ReadSeekCloser which we'd need.

Reply:
That's unfortunate.  Maybe we need a `streaming.LimitReader` that's an `io.ReadSeeker` at some point.